solc/CommandLineParser.cpp

```diff
@@ -898,13 +898,24 @@ bool CommandLineParser::processArgs()
 	else
 		m_options.input.mode = InputMode::Compiler;
 
-	if (
-		m_args.count(g_strExperimentalViaIR) > 0 &&
-		m_options.input.mode != InputMode::Compiler &&
-		m_options.input.mode != InputMode::CompilerWithASTImport
-	)
+	map<string, set<InputMode>> validOptionInputModeCombinations = {
+		// TODO: This should eventually contain all options.
+		{g_strErrorRecovery, {InputMode::Compiler, InputMode::CompilerWithASTImport}},
+		{g_strExperimentalViaIR, {InputMode::Compiler, InputMode::CompilerWithASTImport}},
+		{g_strMetadataLiteral, {InputMode::Compiler, InputMode::CompilerWithASTImport}},
+		{g_strModelCheckerShowUnproved, {InputMode::Compiler, InputMode::CompilerWithASTImport}},
+		{g_strModelCheckerDivModNoSlacks, {InputMode::Compiler, InputMode::CompilerWithASTImport}}
```

Comment:
So what about this? Can you add all model checker options? I think we should do them all in one go.

Reply:
@nishant-sachdeva 
> Since some of the model-checker options have different values for the flags, and we may not need to disallow all of the values, is there some compiled list / documentation which mentions what options are allowed for which compiler mode ?
> Or do have to manually go through the code and check that ?

If you want the values allowed on the CLI see `solc --help` or [this part of `CommandLineParser::optionsDescription()`](https://github.com/ethereum/solidity/blob/v0.8.14/solc/CommandLineParser.cpp#L785-L834).

Same info for Standard JSON is under [Compiler Input and Output JSON Description > Input description](https://docs.soliditylang.org/en/latest/using-the-compiler.html#input-description). Look at the `modelChecker` field.

And [SMTChecker and Formal Verification](https://docs.soliditylang.org/en/latest/smtchecker.html) has an in-depth explanation of what these options do.

Reply:
Pushed code to resolve this comment.